nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_constructor.rs

```diff
@@ -62,7 +63,10 @@ impl StringConstructor {
             // 2. Else,
             // a. If NewTarget is undefined and value is a Symbol, return SymbolDescriptiveString(value).
             if new_target.is_none() && value.is_symbol() {
```

Comment:
**suggestion:** This could be:
```rs
if let Some(Value::Symbol(value)) = value {
  return Ok(Symbol::from(value).descriptive_string(agent).into_value());
}
```

And actually, it might not be a bad idea to change `Value::Symbol(SymbolIndex)` into a `Value::Symbol(Symbol)`. I started thinking yesterday that that's probably going to be what we should do.